client/components/modal/modal.service.js

```diff
@@ -207,6 +207,32 @@ app.factory('Modal', function($rootScope, $uibModal) {
         },
         model: {}
       }, 'modal-success', 'components/modal/form-modal.html');
+    },
+
+    confirmDelete: function(title, templateUrl, model, cb) {
+      var confirmDelete = openModal({
+        modal: {
+          dismissable: true,
+          title: title,
+          templateUrl: templateUrl,
+          submitFn: function(form, model) {
+            cb(model)
+            confirmDelete.close();
+          },
```

Comment:
Yes but then the `cb` function has to return a promise. If we decide to make this change the student.controller.js needs to be updated as well.